src/utils/ticket/YouTrackTicketStatusFetcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils\ticket;
+
+use RuntimeException;
+use staabm\PHPStanTodoBy\utils\CredentialsHelper;
+use staabm\PHPStanTodoBy\utils\HttpClient;
+use function array_key_exists;
+use function is_array;
+use function is_null;
+
+final class YouTrackTicketStatusFetcher implements TicketStatusFetcher
+{
+    private string $host;
+    private ?string $authorizationHeader;
+
+    private HttpClient $httpClient;
+
+    /**
+     * @var array<string, ?string>
+     */
+    private array $cache;
+
+    public function __construct(string $host, ?string $credentials, ?string $credentialsFilePath, HttpClient $httpClient)
```

Comment:
for discussion: in jira we use issue-states to tell when/whether a ticket is resolved. can/should we do the same for youtrack? does youtrack also have tickets flow thru different states?

Reply:
Youtrack uses columns to track the state. 

From the API documentation the resolved field of the `Issue` [[1](https://www.jetbrains.com/help/youtrack/devportal/api-entity-Issue.html)] Entity is explained as
>The timestamp of the moment when the issue was assigned a state that is considered to be resolved. null if the issue is still in an unresolved state. Read-only. Can be null.

There is something called `IssueCustomField` [[2](https://www.jetbrains.com/help/youtrack/devportal/api-entity-IssueCustomField.html)] which allows you to track custom fields such as state of a ticket. But this is specific to each setup and isn't being used in our setup. So adding support for this is challenging. I would rather stick to the definition of YouTrack itself and it people really need to use custom field support creating a PR to extend the youtrack integration can be done.